invenio_files_rest/views.py

```diff
@@ -594,6 +594,17 @@ def delete_object(self, bucket, obj, version_id):
             db.session.commit()
             if obj.file_id:
                 remove_file_data.delay(str(obj.file_id))
+            # Set newest object as head
+            if obj.is_head:
+                try:
+                    obj_to_restore = \
+                        ObjectVersion.get_versions(obj.bucket,
+                                                   obj.key,
+                                                   desc=True).first()
+                    obj_to_restore.is_head = True
+                    db.session.commit()
```

Comment:
I think the entire operation should be atomic thus, there should only be one ``db.session.commit()`` in the method, and the ``remove_file_data`` task should be called after the commit.